bundles/org.openhab.core.io.http.auth/src/main/java/org/openhab/core/io/http/auth/internal/AuthorizePageServlet.java

```diff
@@ -68,11 +70,14 @@
 
     private final Logger logger = LoggerFactory.getLogger(AuthorizePageServlet.class);
 
-    private HashSet<String> csrfTokens = new HashSet<>();
+    private HashMap<String, Instant> csrfTokens = new HashMap<>();
 
     private HttpService httpService;
     private UserRegistry userRegistry;
     private AuthenticationProvider authProvider;
+    @Nullable
+    private Instant lastAuthenticationFailure;
```

Comment:
Hm, theoretically, you would need to keep track of this per user - otherwise, other users would not be able to login if a certain user forgot his password and did too many tries.
For our openHAB situation, it could be ok to block log in in general, though as we are not talking about many different user accounts.

Reply:
That's what I thought too, to keep things simple. As a "last resort" restarting the bundle (or openHAB) does reset the timer too.